### PR TITLE
fix(target): Incorrectly allowing whitespace on Target's address field.

### PR DESCRIPTION
**Pull request by hugoamvieira**

### CODE REVIEW

**Review by johanbrandhorst — APPROVED** [review]
Is this a regression or was it always possible? Is this normally where we make this coercion? I kinda would've expected it to be at the handler level.

**Comment by hugoamvieira** [comment]
@johanbrandhorst 

> Is this a regression or was it always possible?

This was always possible as we missed this case.

> Is this normally where we make this coercion? I kinda would've expected it to be at the handler level.

For the Hosts, we have this check on the repository itself (before it goes into the DB), however @tmessi and I thought it makes more sense on the constructor of the thing you're trying to create.

File: internal/target/registry.go [hunk 1]
```diff
@@ -163,7 +164,14 @@ func New(ctx context.Context, subtype subtypes.Subtype, projectId string, opt ..
...
 		return nil, errors.New(ctx, errors.InvalidParameter, op, "unsupported subtype")
 	}
-	return nf(projectId, opt...)
+
+	t, err := nf(projectId, opt...)
+	if err != nil {
+		return nil, errors.New(ctx, errors.Internal, op, "failed to create target")
+	}
+
+	t.SetAddress(strings.TrimSpace(t.GetAddress()))
```
[the remark this answers is not in the document]
Comment (hugoamvieira) [reply internal/target/registry.go]: Addressed @talanknight

**Review by johanbrandhorst — APPROVED** [review]
LGTM, I believe this should address Todd's concerns too.

**Pull request merged**